### PR TITLE
[Mac OS] add support native OpenAl for mac os 

### PR DESCRIPTION
**Pull request by pperehozhih**
[Mac OS] add support native OpenAl for mac os
(need add cmake arg PROCXX_USE_NATIVE_OPEN_AL)

### CODE REVIEW

**Comment by crackedmind** [comment]
:white_check_mark: [Build kepka 1.0.57-fclmtcst completed](https://ci.appveyor.com/project/procxx/tdesktop/build/1.0.57-fclmtcst) (commit https://github.com/procxx/kepka/commit/2bc2cc55e1 by @pperehozhih)

File: Telegram/SourceFiles/media/media_defs.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+
+/*
+This file is part of Kepka,
+the unofficial desktop version of Telegram messaging app, see https://procxx.github.io/
```
Comment (ForNeVeR) [inline Telegram/SourceFiles/media/media_defs.h]: `the` → `an`? *!!*

File: Telegram/SourceFiles/media/media_defs.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Telegram Desktop is free software: you can redistribute it and/or modify
```
Comment (ForNeVeR) [inline Telegram/SourceFiles/media/media_defs.h]: `Telegram Desktop` → `Kepka`

**Comment by crackedmind** [comment]
:white_check_mark: [Build kepka 1.0.59-yhoimuro completed](https://ci.appveyor.com/project/procxx/tdesktop/build/1.0.59-yhoimuro) (commit https://github.com/procxx/kepka/commit/d16c41963c by @pperehozhih)

File: Telegram/SourceFiles/media/media_defs.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
+
```
Comment (berkus) [inline Telegram/SourceFiles/media/media_defs.h:1]: Remove empty line 1

**Review by zamazan4ik — APPROVED** [review]

File: CMakeLists.txt [hunk 4]
```diff
@@ -44,7 +44,10 @@ if (LINUX)
     find_package(ALSA)
     find_package(PulseAudio)
     find_package(Qt5 COMPONENTS DBus)
-else(LINUX)
+elseif(APPLE AND PROCXX_USE_NATIVE_OPEN_AL)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -framework OpenAL")
```
Comment (berkus) [inline CMakeLists.txt:48]: This is a wrong location, should be in Telegram/CMakeLists.txt where all frameworks are added.

File: Telegram/CMakeLists.txt [hunk 5]
```diff
@@ -5,6 +5,10 @@ if (LINUX)
     else()
         add_definitions(-DQ_OS_LINUX32)
     endif()
+elseif(APPLE)
+    if (PROCXX_USE_NATIVE_OPEN_AL_MAC)
```
Comment (berkus) [inline Telegram/CMakeLists.txt:9]: drop PROCXX_USE_NATIVE_OPEN_AL_MAC check, use native OpenAL on mac unconditionally.

File: Telegram/CMakeLists.txt [hunk 6]
```diff
@@ -5,6 +5,10 @@ if (LINUX)
...
+        add_definitions(-DPROCXX_USE_NATIVE_OPEN_AL_MAC)
```
Comment (berkus) [inline Telegram/CMakeLists.txt:10]: Use `target_compile_definitions()` instead below in the target section

**Review by berkus — CHANGES_REQUESTED** [review]

**Comment by berkus** [comment]
Based on the discussion and usage of ALC_CONNECTED in Kepka sources (which is not supported by MacOS OpenAL) I am not sure this PR can be merged as-is - it will probably break device disconnection scenarios.

The only reason proposed for this PR is "less dependencies" but on mac `brew install openal-soft` is the simplest method to obtain the dependency so I don't see it as a problem. Conan packages also have some openal-soft builds.

Links related to discussion:
- http://openal.996291.n3.nabble.com/Defective-ALC-ENUMERATION-EXT-support-on-Mac-OS-X-Need-OpenAL-community-s-support-td1135.html
- http://icculus.org/alextreg/wiki/action=printer&id=ALC_EXT_Disconnect

**Comment by berkus** [comment]
Lets close this for now, as there seem to be no acceptable changes to do.

**Pull request closed without merging**